### PR TITLE
Start wmde.11 release

### PR DESCRIPTION
**Pull request by addshore**
Makes a wmde.11 release using MediaWiki 1.39
But it also includes various other things to get there...
 - Switched to using `docker.elastic.co` images for elastic 
 - Ignore E_DEPRECATED errors throughout PHP apps
 - Fixes to browser tests needed due to upgrade
 - "Hack" to use wfLoadExtension in tests when upgrading

### CODE REVIEW

**Comment by addshore** [comment]
```
  | <b>Deprecated</b>:  Wikibase\DataModel\Entity\ItemId implements the Serializable interface, which is deprecated. Implement __serialize() and __unserialize() instead (or in addition, if support for old PHP versions is necessary) in <b>/var/www/html/extensions/Wikibase/lib/packages/wikibase/data-model/src/Entity/ItemId.php</b> on line <b>12</b><br />
  | <br />
  | <b>Deprecated</b>:  Wikibase\DataModel\Entity\NumericPropertyId implements the Serializable interface, which is deprecated. Implement __serialize() and __unserialize() instead (or in addition, if support for old PHP versions is necessary) in <b>/var/www/html/extensions/Wikibase/lib/packages/wikibase/data-model/src/Entity/NumericPropertyId.php</b> on line <b>11</b><br />
  | <br />
  | <b>Deprecated</b>:  DataValues\StringValue implements the Serializable interface, which is deprecated. Implement __serialize() and __unserialize() instead (or in addition, if support for old PHP versions is necessary) in <b>/var/www/html/vendor/data-values/data-values/src/StringValue.php</b> on line <b>7</b><br />
  | <br />
  | <b>Deprecated</b>:  DataValues\UnknownValue implements the Serializable interface, which is deprecated. Implement __serialize() and __unserialize() instead (or in addition, if support for old PHP versions is necessary) in <b>/var/www/html/vendor/data-values/data-values/src/UnknownValue.php</b> on line <b>0</b><br />
  | <br />
  | <b>Deprecated</b>:  DataValues\Geo\Values\GlobeCoordinateValue implements the Serializable interface, which is deprecated. Implement __serialize() and __unserialize() instead (or in addition, if support for old PHP versions is necessary) in <b>/var/www/html/vendor/data-values/geo/src/Values/GlobeCoordinateValue.php</b> on line <b>20</b><br />
  | <br />
  | <b>Deprecated</b>:  DataValues\MonolingualTextValue implements the Serializable interface, which is deprecated. Implement __serialize() and __unserialize() instead (or in addition, if support for old PHP versions is necessary) in <b>/var/www/html/vendor/data-values/common/src/DataValues/MonolingualTextValue.php</b> on line <b>0</b><br />
  | <br />
  | <b>Deprecated</b>:  DataValues\UnboundedQuantityValue implements the Serializable interface, which is deprecated. Implement __serialize() and __unserialize() instead (or in addition, if support for old PHP versions is necessary) in <b>/var/www/html/vendor/data-values/number/src/DataValues/UnboundedQuantityValue.php</b> on line <b>23</b><br />
  | <br />
  | <b>Deprecated</b>:  DataValues\QuantityValue implements the Serializable interface, which is deprecated. Implement __serialize() and __unserialize() instead (or in addition, if support for old PHP versions is necessary) in <b>/var/www/html/vendor/data-values/number/src/DataValues/QuantityValue.php</b> on line <b>23</b><br />
  | <br />
  | <b>Deprecated</b>:  DataValues\TimeValue implements the Serializable interface, which is deprecated. Implement __serialize() and __unserialize() instead (or in addition, if support for old PHP versions is necessary) in <b>/var/www/html/vendor/data-values/time/src/DataValues/TimeValue.php</b> on line <b>15</b><br />
 
```

**Comment by addshore** [comment]
Found a task which said it is resolved https://phabricator.wikimedia.org/T301249

**Comment by addshore** [comment]
This doesnt need to be tackled, but ignoring the deprecations in tests would need to happen

**Comment by addshore** [comment]
So there are patches to fix these deprecation warnings that we can backport, but currently, they are not in the REL1_39 branch

Backporting 5 changes to REL1_39 sounds easy enough...

Not in any release branch yet:
 - https://gerrit.wikimedia.org/r/c/mediawiki/extensions/Wikibase/+/845589 & https://gerrit.wikimedia.org/r/c/mediawiki/extensions/Wikibase/+/848265
   - DataValues\StringValue
   - DataValues\UnknownValue 
   - DataValues\Geo\Values\GlobeCoordinateValue
   - DataValues\MonolingualTextValue
   - DataValues\UnboundedQuantityValue
   - DataValues\QuantityValue
   - DataValues\TimeValue
 - FederatedPropertyId https://gerrit.wikimedia.org/r/c/mediawiki/extensions/Wikibase/+/850189/ 
 - EntityId https://gerrit.wikimedia.org/r/c/mediawiki/extensions/Wikibase/+/853286/
 - https://gerrit.wikimedia.org/r/c/mediawiki/extensions/Wikibase/+/773521
   - Wikibase\DataModel\Entity\ItemId https://gerrit.wikimedia.org/r/c/mediawiki/extensions/Wikibase/+/773521/7/lib/packages/wikibase/data-model/src/Entity/ItemId.php
   - Wikibase\DataModel\Entity\NumericPropertyId https://gerrit.wikimedia.org/r/c/mediawiki/extensions/Wikibase/+/773521/7/lib/packages/wikibase/data-model/src/Entity/NumericPropertyId.php

Already in 1.38, 1.39 and above
 - RepoRevisionIdentifier.php & RepoRevisionIdentifier.php  https://gerrit.wikimedia.org/r/c/mediawiki/extensions/Wikibase/+/762094

**Comment by addshore** [comment]
So the needed backports would be....

remote:   https://gerrit.wikimedia.org/r/c/mediawiki/extensions/Wikibase/+/880548 FederatedPropertyId: Add __serialize()/__unserialize() [NEW]
remote:   https://gerrit.wikimedia.org/r/c/mediawiki/extensions/Wikibase/+/880549 Add __serialize() and __unserialize() to EntityId [NEW]
remote:   https://gerrit.wikimedia.org/r/c/mediawiki/extensions/Wikibase/+/880550 DataModel\Entity: Add __serialize/__unserialize for EntityIds [NEW]
remote:   https://gerrit.wikimedia.org/r/c/mediawiki/extensions/Wikibase/+/880551 Drop compatibility with older data-values versions [NEW]

**Comment by addshore** [comment]
In order to backport these, I would also need to backport the other fixes to other extensions such as leeme

From https://integration.wikimedia.org/ci/job/quibble-composer-mysql-php74-noselenium-docker/31167/console
I could do this, but will press pause for now
```
23:15:30 PHP Fatal error:  Class Wikibase\Lexeme\Domain\Model\LexemeId contains 2 abstract methods and must therefore be declared abstract or implement the remaining methods (Wikibase\DataModel\Entity\EntityId::__serialize, Wikibase\DataModel\Entity\EntityId::__unserialize) in /workspace/src/extensions/WikibaseLexeme/src/Domain/Model/LexemeId.php on line 18
23:15:30 
23:15:30 Fatal error: Class Wikibase\Lexeme\Domain\Model\LexemeId contains 2 abstract methods and must therefore be declared abstract or implement the remaining methods (Wikibase\DataModel\Entity\EntityId::__serialize, Wikibase\DataModel\Entity\EntityId::__unserialize) in /workspace/src/extensions/WikibaseLexeme/src/Domain/Model/LexemeId.php on line 18
```

**Comment by addshore** [comment]
So, we should backport to all Wikibase extensions at once to make the release branches all play nice together

So for Lexeme that is also: https://gerrit.wikimedia.org/r/c/840102 https://gerrit.wikimedia.org/r/c/mediawiki/extensions/WikibaseLexeme/+/840101/

For quality constraints that is https://gerrit.wikimedia.org/r/c/mediawiki/extensions/WikibaseQualityConstraints/+/845590

For media info that is https://gerrit.wikimedia.org/r/c/mediawiki/extensions/WikibaseMediaInfo/+/769402 & https://gerrit.wikimedia.org/r/c/mediawiki/extensions/WikibaseMediaInfo/+/842434/

Thats all I can find so lets see if it goes green!

**Comment by addshore** [comment]
Looks like most things are coming up green now which is nice
I am guessing (can't confirm) that perhaps the upgrade tests are failing as they are upgrading with the old local settings files, which still have the old loading of entry points,
So probably need to fix this in CI somehow and note it in the release?

**Comment by addshore** [comment]
Upgrade tests appear to be fixed now after more poking,
Now I just need to go through and debug the individual test failures.

**Comment by addshore** [comment]
So failing jobs now are....

 - Build and test Wikibase and friends / test_wikibase (mariadb:10.9, fedprops) (pull_request) Failing after 6m
 - Build and test Wikibase and friends / test_wikibase (mariadb:10.9, repo_client) (pull_request) Failing after 5m
 - Build and test Wikibase and friends / test_wikibase (mariadb:10.9, quickstatements) (pull_request) Cancelled after 20m
 - Build and test Wikibase and friends / test_wikibase (mariadb:10.9, elasticsearch) (pull_request) Failing after 4m
 - Build and test Wikibase and friends / test_wikibase (mariadb:10.9, base__fedprops) (pull_request) Failing after 5m

Everything federated properties related seems to fail as deletions of statements timeout in test setup?
repo_client seems to be something relating to dispatching (again)....
quickstatements timed out waiting for things... This is odd, as this is the same version, perhaps a 1 off failure? I'll re run them....
elastic search just fails while searching `should be able to search case-insensitive`

Will re run and see if anything goes away

**Comment by addshore** [comment]
After a re run the only issues appear to be 1) fed prosp 2) quick statements 3) elastic search

Having already briefly investigated fed props and determined deleting statements with fed props on is slow, I'll investigate the other 2 a little.

**Comment by addshore** [comment]
Quickstatements issue appears to be caused by

```
<br />
<b>Fatal error</b>:  Composer detected issues in your platform: Your Composer dependencies require a PHP version ">= 8.1.0". You are running 7.3.33. in <b>/var/www/html/quickstatements/vendor/composer/platform_check.php</b> on line <b>24</b><br />
```

So composer deps are installed with 8.1 bu the image still uses 7.3?
This will have been caused when the composer version was bumped in https://github.com/wmde/wikibase-release-pipeline/pull/398/commits/74622cf63d821769caea7a5f54662cf6d4435821
Unknowingly this also bumped the version used for quickstatements...

**Comment by addshore** [comment]
Next steps...

There is no 7.10.2 tag for elasticsearch on dockerhub
image

The OAuth maint script that is used is broken in 1.39
https://gerrit.wikimedia.org/r/c/mediawiki/extensions/OAuth/+/887828 & backport are up

**Comment by addshore** [comment]
It looks like the docker hub and docker.elastic.co images are the same, so I'll switch away from docker hub

image

**Comment by addshore** [comment]
Elastic search is now green for CI.
The 2 remaining issue are fed props and quick statements.
Quickstatements needs an OAuth fix, and fed props is still an unknonw issue.

**Comment by addshore** [comment]
OAuth changes are merged now, so a rebuild should only lead to federated properties tests breaking...

**Comment by addshore** [comment]
quickstatements failed again, I'll have to debug it locally once more...

**Comment by addshore** [comment]
Looking at the last build 

```
Checking out OAuth at
commit 8481fe07b4ae1939b629fc7f9195e820aa84efdc
Merge: 055fcd58 4949490b
Author: jenkins-bot <jenkins-bot@gerrit.wikimedia.org>
Date:   Fri Feb 10 22:15:31 2023 +0000

    Merge "maint/createOAuthConsumer.php: Specify oauth2GrantTypes" into REL1_39
```
So the fix is in, lets debug locally with this build now...

**Comment by addshore** [comment]
image

**Comment by addshore** [comment]
Quickstatements then also runs into deprecation warning issues...

```
<br />
<b>Deprecated</b>:  Implicit conversion from float 0.1 to int loses precision in <b>/var/www/html/quickstatements/public_html/quickstatements.php</b> on line <b>1126</b><br />
{"status":"OK","command":{"action":"create","type":"item","meta":{"message":"","status":"RUN","id":0},"summary":"#temporary_batch_1676476421680","status":"done","run":{"action":"wbeditentity","new":"item","data":"{}","summary":"#quickstatements; #temporary_batch_1676476421680","bot":1,"token":"c804fdc492865f4faeec08656f11853f63ed000c+\\","format":"json"},"item":"Q4"},"last_item":"Q4"}
```

**Comment by addshore** [comment]
https://github.com/magnusmanske/quickstatements/blob/f3d5ff092284949a9880747f1259b343f62a2ce9/public_html/vue_components/batch.html#L358

```
Uncaught ReferenceError: d is not defined
```

**Comment by addshore** [comment]
Will wait and see what happens in https://github.com/magnusmanske/quickstatements/pull/43

**Comment by addshore** [comment]
Looks like quickstatements is fixed, the only remaining issue looks to be the feddy props tests and https://phabricator.wikimedia.org/T329308

**Comment by addshore** [comment]
feddy props tests fixed, and a final update of various versions done, should be green when CI finishes!

File: Docker/test/selenium/specs/fedprops/prefetching.js [hunk 1]
```diff
@@ -59,21 +59,28 @@ describe( 'Property Prefetching', function () {
...
 	} );
 
 	it( 'Should render recent changes list within threshold', function () {
 		browser.url( process.env.MW_SERVER + '/wiki/Special:RecentChanges?limit=50&days=7&urlversion=2&enhanced=0' );
 		$( 'ul.special' ).waitForDisplayed( { timeout: 2000 } );
 
-		assert( $$( 'ul.special li' ).length >= NUM_PROPERTIES );
+		// +1 for the initial item creation
+		// +1 for the Main Page creation?
+		// +1 for unknown reason?
```
Comment (manicki) [inline Docker/test/selenium/specs/fedprops/prefetching.js:82]: well, that's great

**Review by manicki — COMMENTED** [review]
It would be perfect if the `debug_extra_env` thing and github codespaces docs could go to a separate PR.
Otherwise looks alright, although I still need to look at the shell script changes (only skimmed through them so far)

File: versions/wmde10.env [hunk 2]
```diff
@@ -10,9 +10,9 @@ WDQS_FRONTEND_COMMIT_HASH=fb1e5bfc12491a41e45389a613816967c302dd43
 QUICKSTATEMENTS_COMMIT_HASH=a2d155f5f90ca7ba26ebd6ce7f2892de9999961a
 # Head of master on 03/01/2023
 MAGNUSTOOLS_COMMIT_HASH=68c34ee04d3c6d68fb9e2106821f9686a96d2075
-# 1.0.2 latest release on 03/01/2023
+# 1.0.2 latest release on 03/01/2023 (doesn't state 1.38 compatability)
```
Comment (tarrow) [inline versions/wmde10.env:13]: so does it work properly?

Comment (addshore) [reply versions/wmde10.env:13]: The tests pass, that is all I check as part of the creation of this PR
There are also product tests that will happen once this PR is merged.
This comment is only to help devs in the future if they glance at the commit hash and want to know what that actually means

**Review by manicki — APPROVED** [review]
pull request contains a few changes which are not immediately obvious and does not seem there's loads of explanation provided (e.g. change in the selenium test, composer related hack in the build_wikibase script)
I guess it is what it is

**Pull request merged**